### DISCUSSION TITLE
Passing sceneInstance to "Engine.director.getSceneName" does not work it.

### DISCUSSION THREAD

**Discussion by aruru-weed** [excaliburjs/Excalibur#3021] [Q&A]
Hi! I was thinking of making a game using TS for the first time, and was excited to find Excalibar.js.

I'm not good at trying to manage Scenes with strings, so I use "game.goToScene(game.director.getSceneName(scene))"
When I tried that, it didn't work.

![image](https://github.com/excaliburjs/Excalibur/assets/58074520/8a4cfa18-607f-409a-8b69-11ab63478be1)

However, I discovered that by running "game.director.getSceneInstance("MapName")" before it, it works fine.

![image](https://github.com/excaliburjs/Excalibur/assets/58074520/cfd72a51-dd33-4938-b22d-b30bd6784278)

Is this nominal? If it's a bug, I'll try my best to send a pull request.

**Comment by eonarheim** [comment]
@aruru-weed Good question, definitely seems like a bug! I'd expect it to work the way you want.

My guess is there is a cache being populated in the 2nd case.

**Reply by eonarheim** [reply]
Made a bug! https://github.com/excaliburjs/Excalibur/issues/3022
